<?xml version="1.0" encoding="UTF-8"?>

<launch>
    <arg name="aruco_dictionary" default="DICT_4X4_50"/>
    <arg name="aruco_id" default="0"/>
    <arg name="x" default="0.0" />
    <arg name="y" default="0.0" />
    <arg name="z" default="0.0" />
    <arg name="roll" default="0"/>
    <arg name="pitch" default="0"/>
    <arg name="yaw" default="0.0" />

    <include file="$(find-pkg-share aruco_description)/launch/load_aruco_description.launch">
        <arg name="aruco_dictionary" value="$(var aruco_dictionary)"/>
        <arg name="aruco_id" value="$(var aruco_id)"/>
        <arg name="aruco_description_name" value="$(var aruco_dictionary)_id$(var aruco_id)_description"/>
    </include>

    <node name="spawn_aruco" pkg="gazebo_ros" exec="spawn_entity.py" output="screen" args="-urdf -param $(var aruco_dictionary)_id$(var aruco_id)_description -model $(var aruco_dictionary)_id$(var aruco_id) -x $(var x) -y $(var y) -z $(var z) -R $(var roll) -P $(var pitch) -Y $(var yaw)" />
</launch>